component/src/main/java/io/siddhi/extension/io/grpc/sink/GrpcSink.java

```diff
@@ -0,0 +1,347 @@
+                        "      @map(type='json'),\n" +
+                        "           @payload('{{message}}'))\n" +
+                        "define stream FooStream (message String, headers String);",
+                        description = "A similar example to above but with headers. Headers are also send into the " +
+                                "stream as a data. In the sink headers dynamic property reads the value and sends " +
+                                "it as MetaData with the request"
+                ),
+                @Example(syntax = "" +
+                        "@sink(type='grpc',\n" +
+                        "      publisher.url = 'grpc://134.23.43.35:8080/org.wso2.grpc.test.MyService/send',\n" +
+                        "      @map(type='protobuf'),\n" +
+                        "define stream FooStream (stringValue string, intValue int,longValue long,booleanValue bool," +
+                        "floatValue float,doubleValue double);",
+                        description = "Here a stream named FooStream is defined with grpc sink.A grpc server" +
+                                "should be running at 194.23.98.100 listening to port 8080, since there is no mapper " +
+                                "provided, attributes of stream definition should be same as the attributes of " +
+                                "protobuf message definition"
+                )
+        }
+)
+
+public class GrpcSink extends AbstractGrpcSink {
+    private static final Logger logger = Logger.getLogger(GrpcSink.class.getName());
+    private AbstractStub asyncStub;
+
+    @Override
+    public void initSink(OptionHolder optionHolder) {
+        if (isDefaultMode) {
+            if (methodName == null) {
+                methodName = GrpcConstants.DEFAULT_METHOD_NAME_WITHOUT_RESPONSE;
+            } else if (!methodName.equalsIgnoreCase(GrpcConstants.DEFAULT_METHOD_NAME_WITHOUT_RESPONSE)) {
+                throw new SiddhiAppValidationException(siddhiAppName + ": " + streamID + ": In default " +
+                        "mode grpc-sink when using EventService the method name should be '" +
+                        GrpcConstants.DEFAULT_METHOD_NAME_WITHOUT_RESPONSE + "' but given " + methodName);
+            }
+        }
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions dynamicOptions, State state)
+            throws ConnectionUnavailableException {
+        StreamObserver<Empty> responseObserver = new StreamObserver<Empty>() {
+            //try to send all the siddhi events using one stream observer - impossible without adding client
+            // side streaming in protobuf definition
+            @Override
+            public void onNext(Empty event) {
+            }
+
+            @Override //todo latch based error???
+            public void onError(Throwable t) { //parent method doest have error in its signature. so cant throw
+                // from here
```

Comment:
@niruhan could please remove these unwanted comments

Reply:
Fixed in https://github.com/siddhi-io/siddhi-io-grpc/pull/6